### PR TITLE
[improve][transform] flink engine support flink-sql、udf components

### PR DESCRIPTION
**Pull request by YMBSKLK**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java [hunk 1]
```diff
@@ -125,60 +129,63 @@ public PreparedStatement toExternal(
...
-                    statement.setNull(statementIndex, java.sql.Types.NULL);
-                    break;
-                case MAP:
-                case ARRAY:
-                case ROW:
-                default:
-                    throw new JdbcConnectorException(
-                            CommonErrorCode.UNSUPPORTED_DATA_TYPE,
-                            "Unexpected value: " + seaTunnelDataType);
+            Object obj = row.getField(fieldIndex);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java:129]: Why change it?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java [hunk 2]
```diff
@@ -80,6 +85,10 @@ public void write(@NonNull SeaTunnelRow seaTunnelRow) {
         FSDataOutputStream fsDataOutputStream = getOrCreateOutputStream(filePath);
         try {
             if (isFirstWrite.get(filePath)) {
+                if(isPrintHeader){
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: Why add this?

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/transform/SeaTunnelTransform.java [hunk 3]
```diff
@@ -26,10 +26,7 @@
 import java.io.Serializable;
 
 public interface SeaTunnelTransform<T>
-        extends Serializable,
-                PluginIdentifierInterface,
-                SeaTunnelPluginLifeCycle,
-                SeaTunnelJobAware {
+        extends Transform {
```
Comment (TyrantLucifer) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/transform/SeaTunnelTransform.java]: Do you have a disscuss with community?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Thank you for your contribution, but the function of this pull request I think you should start a disscuss with community and write a proposal. cc @hailin0 @EricJoy2048

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java (hunk 1 above)
Comment (YMBSKLK) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/converter/AbstractJdbcRowConverter.java:129]: When the field content is empty, outputting to a relational database will throw an exception, so it is better to check if it is empty and set it to NULL directly.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java (hunk 2 above)
Comment (YMBSKLK) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: When exporting data to CSV or TXT, if the field names are not exported, it is often necessary to export the field names.

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/transform/SeaTunnelTransform.java (hunk 3 above)
Comment (YMBSKLK) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/transform/SeaTunnelTransform.java]: I'm sorry, this change was not discussed, but I believe Flink SQL can solve many complex data transformation scenarios.

**Comment by hailin0** [comment]
> Thank you for your contribution, but the function of this pull request I think you should start a disscuss with community and write a proposal. cc @hailin0 @EricJoy2048

+1

**Comment by EricJoy2048** [comment]
+1

**Comment by EricJoy2048** [comment]
We have spent a long time in the past six months removing the dependencies between Flink SQL and Spark SQL, and implementing SeaTunnel's own SQL Transform, which is designed to run on both Flink/Spark/Zeta. I don't think adding Flink SQL back is a good way.

Anyway, an open discussion is necessary. You can post your proposal on the dev. seatunnel. apache. org mailing list, and the community will discuss it and ultimately determine whether the proposal is accepted

**Comment by TyrantLucifer** [comment]
I will close it first, if you have any advice, please start a discussion in `dev@seatunnel.apache.org`.

**Pull request closed without merging**